docs/api-docs/in-depth-guides/rail-integration-guide.mdx

```diff
@@ -1,100 +1,167 @@
 
+This is a more complex diagram that shows the various events that can occur in a container's rail journey.
 
 ```mermaid
 graph LR
+    Y[From ocean terminal or carrier] --> A
+    Z[Rail Interchange Received] --> B
     A[Rail Loaded] --> B[Rail Departed]
+    B --> L[Train Passing]
     B --> C[Rail Arrived]
+    L --> L
+    L --> C
     C --> D[Rail Unloaded]
-    D --> E[Full Out]
+    C --> X[Rail Interchange Delivered]
+    X --> Z
+    D --> G[Available for Pickup]
+    D --> H[Not Available]
+    H -- Holds and Fees Updated --> H
+    G --> H
+    H --> G
+    G --> E[Full Out]
     E --> F[Empty Return]
-  ```
+``` */}
 
-```mermaid
-timeline
-    title Container Milestones (Rail Import)
-    section Port of Discharge
-        Vessel Arrival : Port of discharge : Terminal details : discharged timestamp
-        Vessel Birth : Port of discharge : Terminal details : Birth timestamp
-        Container Discharged : Port of discharge : Terminal details : discharged timestamp
-    section Container On Rail
-        Rail Loaded : Loaded location : loaded timestamp
-        Rail Departed : Departed location : departed timestamp
-        Rail Arrived : Arrived location : arrived timestamp
-        Rail Unloaded : Unloaded location : unloaded timestamp
-    section Final Destination (Rail Terminal)
-        Full Out : Full out port : Full out terminal : Full out timestamp
-        Empty Return : empty return timestamp
+In addition, there are events that update you on the status of your container at the rail terminal, marking it as available or unavailable.  Some of these events come directly from the carriers, while others we generate when we see the fees and holds on your container change.
 
+```mermaid
+graph LR
+    C[Previous events] --> D[Rail Unloaded]
+    D --> G[Available for Pickup]
+    D --> H[Not Available]
+    G --> H
+    H --> G
+    H -- Holds and Fees Updated --> H
+    G --> E[Full Out]
 ```
 
-#### Rail Milestones (aka rail events)
+### Webhook Notifications
+
+Terminal49 provides webhook notifications to keep you updated on key Transport Events in a container's rail journey. These notifications allow you to integrate near real-time tracking data directly into your applications.
+
+Here's a list of the rail-specific events which support webhook notifications:
+
+| Transport Event | Webhook Notification | Description |
+|----------------|----------------------|-------------|
+| Rail Loaded    | `container.transport.rail_loaded`   | The container is loaded onto a railcar. |
+| Rail Departed  | `container.transport.rail_departed` | The container departs on the railcar (usually from port of discharge). |
+| Rail Arrived   | `container.transport.rail_arrived`  | The container arrives at the destination terminal. |
```

Comment:
Actually this is `Arrived At Inland Destination' `container.transport.arrived_at_inland_destination`.

Rail Arrived happens at the destination too, but can also happen at stops along the way if the train stops.

Reply:
In that case, I'll add both and make the distinction explicit - they're going to see `rail_arrived` events, don't want them to be confused

Reply:
Done ✅